extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -108,7 +109,17 @@ declare module 'dataworkspace' {
 		/**
 		 * Gets the icon path of the project type
 		 */
-		readonly icon: azdata.IconPath
+		readonly icon: azdata.IconPath;
+
+		/**
+		  * Gets the target platforms that can be selected when creating a new project
+		 */
+		readonly targetPlatforms?: string[];
```

Comment:
this is very specific to SQL project, can we think of a more generic way to provide additional project information?

Reply:
is choosing a project's target platform not something that other project types like pgproj could also use? Or would a term like "version" be better?

Reply:
I don't know for sure, but for these kind of scenarios, maybe it is better to be flexible. for example, add a property called options: projectOption[];

`
interface ProjectOption {
    name: string;
    type: string; // dropdown, string...
}

interface ProjectDropdownOption {
    options:[],
    defaultOption: string;
}
`

and then the dialog can add these options dynamically at runtime.

Reply:
Platform seems pretty generic, if you're building something like projects do there's almost always a platform type parameter you set.

But it also kind of depends on whether you plan on adding more "optional" parameters that a project could have. Because if you do then it seems like it would probably make sense to do what Alan suggested and have there be more of a property bag type thing like resource deployment does. Doing that gets tricky with passing stuff around though - especially when you factor in other APIs such as createProject. Do you just have that take in a property bag of stuff too? How do projects know what other properties other providers are using? Or do you not care since I imagine most extensions are just calling createProject on their own project types so know what they are?

These questions and issues seem similar to the ones we have about connection dialog, might be worth considering how we do it there.

Reply:
I don't think there are any plans in the near future of letting projects add more optional parameters to this dialog, so I'd prefer to keep this simpler for passing the target platform to the createProject API. 

@udeeshagautam @radentonev thoughts on if we should make this more flexible and allow other project types to add more optional parameters?

Reply:
I am fine with this, we can add support for more generic options in the future if needed.